flink-state-backends/flink-statebackend-changelog/src/main/java/org/apache/flink/state/changelog/restore/ChangelogBackendRestoreOperation.java

```diff
@@ -89,14 +82,11 @@
             ChangelogKeyedStateBackend<?> backend,
             ChangelogStateBackendHandle backendHandle,
             StateChangelogHandleReader<T> changelogHandleReader,
-            ClassLoader classLoader,
-            Map<UUID, Map<Short, StateID>> metadataByBackend)
+            ClassLoader classLoader)
             throws Exception {
+        Map<Short, StateID> stateIds = new HashMap<>();
```

Comment:
The equivalent change should be creating the stateIds within the for loop, otherwise, the stateIds would contain short state id created by other state backends. Though we would update the state id during meta restore, I'm afraid that this might not be safe if we change implementation of updating the meta in the future.

Reply:
As discussed offline:
The loop iterates over the log segments of a single backend (`backendHandle.getNonMaterializedStateHandles()`)
So it would be incorrect to re-create the map inside the loop, because earlier log segments might contain metadata needed for the later ones.
And there should be no data from other backends during the whole loop, because `backendHandle` comes from a single backend.